rfcs/0089-collect-non-source-package-meta.md

```diff
@@ -0,0 +1,117 @@
+---
+feature: collect-non-source-package-meta
+start-date: 2021-03-14
+author: Robert Scott
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: @asymmetric, @aforemny, @alyssais
+shepherd-leader: @asymmetric
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+Collect and maintain a new `meta` attribute in packages allowing users to easily
+identify and manage their preference for binary (more broadly "non-source")
+packages.
+
+# Motivation
+[motivation]: #motivation
+
+Different users have different expectations from a software distribution. We
+acknowledge that much with the collection of license information and the
+existence of the `allowUnfree` nixpkgs option, much as Debian maintains a
+separate `-nonfree` repository.
+
+Similarly, there are a number of different reasons users may have to disfavour
+those packages not built-from-source:
+
+- Transparency: an ever-growing concern with more focus than ever on
+  supply-chain attacks.
+- Malleability: being able to conveniently override packages with patches or an
+  altered build process is a key advantage of Nix, and for nixpkgs maintainers
+  it's not generally possible to backport security fixes to binary packages.
+
+For some users, these concerns are enough to deter them from using Nix entirely.
+
+# Detailed design
+[design]: #detailed-design
+
+Add a new `meta` attribute to non-source-built packages, `sourceProvenance`.
+The value of this attribute being a list of at least one entriy from a
+collection of possibilities maintained in `lib.sourceTypes`. These possibilities
+can be of opaque type, but should have entries to represent at least "built from
```

Comment:
What does opaque type mean in this context?

Reply:
I think I meant "you shouldn't expect any specific behaviour from it other than being able to check for equality". At time of writing I think I was leaving open the possibility of having values which were able to understand their own hierarchy to some degree, though the more I think of it, it would probably be better to embed all notions of hierarchy in the `allowNonSourcePredicate` function to allow customization there too.

Reply:
I would then rather change this to remove the reference to opaqueness, since it's rather.. opaque like this.

Do you agree?